src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -598,38 +598,56 @@ void destroyAllOntimeIntervalThreads(std::string tenant)
 
 /* ****************************************************************************
 *
-* includedEntity -
+* matchEntity -
+*
+* Taking into account null types and id patterns. Note this function is not
+* commutative: en1 is interpreted as the entity to match *in* en2 (i.e.
+* it is assumed that the pattern is in en2)
 */
-bool includedEntity(EntityId en, EntityIdVector* entityIdV) {
+bool matchEntity(EntityId* en1, EntityId* en2)
+{
+  bool idMatch;
+  if (isTrue(en2->isPattern))
+  {
+    regex_t regex;
+    if (regcomp(&regex, en2->id.c_str(), 0) != 0)
+    {
+      LM_W(("Bad Input (error compiling regex: '%s')", en2->id.c_str()));
+      idMatch = false;
+    }
+    if (regexec(&regex, en1->id.c_str(), 0, NULL, 0) == 0)
+    {
+      idMatch = true;
+    }
+    else
+    {
+      idMatch = false;
+    }
```

Comment:
If you initialize idMatch to false, you can save five lines of source code.

Also, if regcomp fails, you probably don't want to execute regexec.
I'd propose to do 

    else if (regexec(&re ...